### ISSUE TITLE
Generalise type stubs from str to Sequence[Token]

### ISSUE THREAD

**Issue by wenkokke** [sighingnow/parsec.py#53]
By the looks of it, the only thing that prevents this parser combinator library from working for arbitrary sequences of tokens, is [the following check](https://github.com/sighingnow/parsec.py/blob/master/src/parsec/__init__.py#L128-L130):
```python
if not isinstance(text, str):
    raise TypeError(
        'Can only parsing string but got {!r}'.format(text))
```
Would it be a welcome addition to generalise the typing stubs and add tests for parsing, e.g., lists of strings or tuples of some enum type?

**Comment by sighingnow** [comment]
I think it should work, but some of those combinators, e.g., `regex` may fail, as they assume the underlying container is a string.

**Comment by sighingnow** [comment]
I could take a try.

**Comment by wenkokke** [comment]
> I think it should work, but some of those combinators, e.g., `regex` may fail, as they assume the underlying container is a string.

Yes, I assume `regex` would need a stricter type, though you might generalise it to `AnyStr`.

**Comment by sighingnow** [comment]
Implemented and demostrated the usage in https://github.com/sighingnow/parsec.py/pull/54/files
